### PR TITLE
feat: add proxy DHCP server

### PR DESCRIPTION
**Pull request by lion7**

### CODE REVIEW

**Comment by smira** [comment]
@lion7 I reworked some code while debugging it to use much better DHCP library. Still doing some cleanups, but it should be ready soon. Thanks for your contribution!

**Comment by lion7** [comment]
Rebased this on the latest `main`

**Comment by smira** [comment]
I have it working now in my local environment, need a couple of patches to make things work in QEMU environment. Networking is hard! :)

**Comment by smira** [comment]
ok, I'll cut Talos v1.6.0-alpha.0 today or tomorrow, I will pull it in here (for bridge networking fixes), and the tests should pass in the CI :crossed_fingers:

File: sfyra/cmd/sfyra/cmd/bootstrap_capi.go [hunk 1]
```diff
@@ -33,9 +33,9 @@ var bootstrapCAPICmd = &cobra.Command{
 
 				RegistryMirrors: options.RegistryMirrors,
 
-				CPUs:   options.BootstrapCPUs,
-				MemMB:  options.BootstrapMemMB,
-				DiskGB: options.BootstrapDiskGB,
+				BootstrapCPUs:   options.BootstrapCPUs,
```
Comment (smira) [inline sfyra/cmd/sfyra/cmd/bootstrap_capi.go:36]: change in `sfyra`: migrate from creating two clusters - one for Sidero Metal (management), another for PXE VMs down to a single cluster: two VMs run Talos + Sidero Metal, others are empty PXE VMs.

**Review by smira — APPROVED** [review]

**Review by frezbo — APPROVED** [review]

**Pull request merged**